addons/godot-next/resources/behavior.gd

```diff
@@ -37,38 +37,52 @@ var _enabled: bool = true setget set_enabled, get_enabled
 
 ##### VIRTUALS #####
 
+# '_awake' name is used to match the convention in Unity's MonoBehaviour class.
+func _awake() -> void:
+	pass
+
+func _on_enable() -> void:
+	pass
+
+func _on_disable() -> void:
+	pass
+
 # Should only override if one wishes to create their own abstract Behaviors
 # By default, the absence of this method is interpreted as a non-abstract type!
 static func is_abstract() -> bool:
 	return true
 
 ##### PUBLIC METHODS #####
 
-# Sets up the owner instance on the Behavior.
-# 'awake' name is used to match the convention in Unity's MonoBehaviour class.
-func awake(p_owner) -> void:
-	owner = p_owner
-
 # Returns an instance of the stored Behavior resource.
 func get_behavior(p_type: Script) -> Behavior:
 	return owner.get_behavior(p_type)
 
 ##### PRIVATE METHODS #####
 
+# Sets up the owner instance on the Behavior.
+func __awake(p_owner) -> void:
+	owner = p_owner
+	_awake()
+
 ##### CONNECTIONS #####
 
 ##### SETTERS AND GETTERS #####
 
-func set_enabled(enable: bool) -> void:
-	_enabled = enable
-	var method = "_add_to_callbacks" if _enabled else "_remove_from_callbacks"
-	owner.call(method, self)
+func set_enabled(p_enable: bool) -> void:
+	if _enabled == p_enable:
+		return
+	_enabled = p_enable
+	#warning-ignore:standalone_expression
+	_on_enable() if p_enable else _on_disable()
+	#warning-ignore:standalone_expression
+	owner._add_to_callbacks() if p_enable else owner._remove_from_callbacks()
 
 func get_enabled() -> bool:
 	return _enabled
 
 #warning-ignore:unused_argument
-func _set_owner(owner) -> void:
+func _set_owner(p_owner) -> void:
```

Comment:
Now that I think about it, isn't it possible that one might want to transfer a component from one CallbackDelegator to another? In that case, we should probably `set_owner` NOT be private. It would also be unfair of us to force people to not be able to use the Behavior class in a different context, so we probably should enforce a CallbackDelegator owner either. It should do something like this I think:

    func set_owner(p_owner) -> void:
        assert p_owner
        owner = p_owner